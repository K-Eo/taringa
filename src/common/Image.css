.Image-wrapper {
  border-radius: 0.25rem;
  border: 1px solid #f0f0f0;
  margin: 0 auto 0 0;
  padding: 0;
}

.Image-placeholder {
  background: #f0f0f0;
  max-width: 100%;
  position: relative;
}

.Image-source {
  left: 0;
  max-width: 100%;
  position: absolute;
  top: 0;
}

.StoryPreview .Image-wrapper {
  border-radius: 0;
  border: none;
  margin: 0 auto !important;
}
