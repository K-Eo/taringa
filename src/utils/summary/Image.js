import React from 'react'
import Summary from './Summary'

class Image extends Summary {
  getPreview() {
    const image = this.getFirstSummaryImage()

    if (image) {
      const { url, width, height } = image

      return (
        <img
          alt={url}
          className="img-fluid"
          height={height}
          src={url}
          width={width}
        />
      )
    }

    return null
  }

  getPreviewKind() {
    return this.summary.summary.images.amount > 0 ? 'image' : null
  }

  getTitle() {
    return this.summary.summary.excerpt || this.summary.title
  }

  getIcon() {
    return 'far fa-image'
  }
}

export default Image
