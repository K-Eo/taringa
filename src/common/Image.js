import React from 'react'
import classNames from 'classnames'

import './Image.css'

const Image = ({ center, children, height, rounded, url, width }) => {
  const wrapperClasses = classNames('Image-wrapper', {
    'Image-wrapper-center': center,
  })

  const imgClasses = classNames('Image-source', {
    rounded: rounded,
  })

  return (
    <div className={wrapperClasses} style={{ maxWidth: width }}>
      <div
        className="Image-placeholder"
        style={{ paddingTop: `${height / width * 100}%` }}
      >
        {children ? (
          children
        ) : (
          <img src={url} alt={url} className={imgClasses} />
        )}
      </div>
    </div>
  )
}

Image.defaultProps = {
  height: 1,
  url: '#IMAGE_URL',
  width: 1,
}

export default Image
