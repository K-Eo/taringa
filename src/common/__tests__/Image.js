import React from 'react'

import Image from '../Image'

describe('Image', () => {
  it('exists', () => {
    expect(Image).toBeDefined()
  })

  it('renders default', () => {
    const wrapper = shallow(<Image />)

    expect(wrapper).toMatchSnapshot()
  })

  it('renders img', () => {
    const props = {
      url: 'https://placehold.it/1000x720',
      height: 720,
      width: 1000,
    }

    const wrapper = shallow(<Image {...props} />)

    expect(wrapper).toMatchSnapshot()
  })

  it('renders custom img', () => {
    const props = {
      url: 'https://placehold.it/1000x720',
      height: 720,
      width: 1000,
    }

    const wrapper = shallow(
      <Image {...props}>
        <div>foobar</div>
      </Image>
    )

    expect(wrapper).toMatchSnapshot()
  })
})
