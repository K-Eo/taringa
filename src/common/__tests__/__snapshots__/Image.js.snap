// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image renders custom img 1`] = `
<div
  className="Image-wrapper"
  style={
    Object {
      "maxWidth": 1000,
    }
  }
>
  <div
    className="Image-placeholder"
    style={
      Object {
        "paddingTop": "72%",
      }
    }
  >
    <div>
      foobar
    </div>
  </div>
</div>
`;

exports[`Image renders default 1`] = `
<div
  className="Image-wrapper"
  style={
    Object {
      "maxWidth": 1,
    }
  }
>
  <div
    className="Image-placeholder"
    style={
      Object {
        "paddingTop": "100%",
      }
    }
  >
    <img
      alt="#IMAGE_URL"
      className="Image-source"
      src="#IMAGE_URL"
    />
  </div>
</div>
`;

exports[`Image renders img 1`] = `
<div
  className="Image-wrapper"
  style={
    Object {
      "maxWidth": 1000,
    }
  }
>
  <div
    className="Image-placeholder"
    style={
      Object {
        "paddingTop": "72%",
      }
    }
  >
    <img
      alt="https://placehold.it/1000x720"
      className="Image-source"
      src="https://placehold.it/1000x720"
    />
  </div>
</div>
`;
